ksql-engine/src/main/java/io/confluent/ksql/statement/ConfiguredStatement.java

```diff
@@ -87,6 +94,12 @@ public KsqlConfig getConfig() {
     return new ConfiguredStatement<>(this, properties);
   }
 
+  public ConfiguredStatement<T> withStatement(
+      final String statementText,
+      final T statement) {
+    return new ConfiguredStatement<>(this, PreparedStatement.of(statementText, statement));
```

Comment:
🤷‍♂️ I'm doing this because of this comment https://github.com/confluentinc/ksql/pull/2436#discussion_r256486412 that suggested it is more future proof to create copy constructors. I don't really care much about boiler plate so let me know which way you prefer and I can make it that way.

Reply:
Damn you for throwing my own comments back in my face ;) 

However, the difference here is that https://github.com/confluentinc/ksql/pull/2436#discussion_r256486412 is a _base_ class method, and allows people to add fields to the base class without having to change all derived classes. Where as in this case its all contained in a single file.  

If the private constructors were used by many different methods, then they'd have worth. As it is, I think a single 'I take all fields' constructor is sufficient for all uses in the class.

But... it's no biggie. It's just a style thing.

Reply:
I think that distinction makes sense! I will for this (and going forward) have copy constructors only for classes with funky inheritance.